psignifit/psignifit.py

```diff
@@ -83,31 +83,36 @@ def psignifit(data: np.ndarray, conf: Optional[Configuration] = None,
         bounds.update(conf.bounds)
     if conf.fixed_parameters is not None:
         for param, value in conf.fixed_parameters.items():
-            bounds[param] = (value, value)
-
+            if value is not None:
+                bounds[param] = (value, value)
     priors = setup_priors(custom_priors=conf.priors, bounds=bounds,
                           stimulus_range=stimulus_range, width_min=width_min, width_alpha=conf.width_alpha,
                           beta_prior=conf.beta_prior, threshold_perc_correct=conf.thresh_PC)
     fit_dict, posteriors, grid = _fit_parameters(data, bounds, priors, sigmoid, conf.steps_moving_bounds,
                                                  conf.max_bound_value, conf.grid_steps)
 
-    grid_values = [grid_value for _, grid_value in sorted(grid.items())]
-    intervals = confidence_intervals(posteriors, grid_values, conf.confP, conf.CI_method)
-    intervals_dict = {param: interval_per_p.tolist()
-                      for param, interval_per_p in zip(sorted(grid.keys()), intervals)}
+    grid_none_ix = tuple(ix for ix, (param, value) in enumerate(sorted(grid.items())) if value is None)
+    grid_params = [param for param, value in grid.items() if value is not None]
+    grid_values = [grid[param] for param in grid_params]
+    intervals = confidence_intervals(np.squeeze(posteriors, axis=grid_none_ix), grid_values, conf.confidence_intervals, conf.CI_method)
+    intervals_dict = {param: intervals[ix].tolist() for ix, param in enumerate(grid_params)}
```

Comment:
is this needed, or for Matlab compatibility?

Reply:
Yes, this is needed. Even non-matlab users might not provide all bounds. This case handels these

Reply:
I removed this